code/modules/mob/living/simple_animal/hostile/hog.dm

```diff
@@ -0,0 +1,367 @@
+
+	status_flags = CANSTUN
+
+	harm_intent_damage = 10
+	melee_damage_lower = 4
+	melee_damage_upper = 8
+	attacktext = "gores"
+	attack_sound = 'sound/weapons/bite.ogg'
+
+	min_oxy = 0
+	max_oxy = 0
+	min_tox = 0
+	max_tox = 0
+	min_co2 = 0
+	max_co2 = 0
+	min_n2 = 0
+	max_n2 = 0
+	minbodytemp = 0
+
+	faction = "HOG"
+
+	nutrition = HOG_FED
+	var/food_search_radius = 7
+	var/food_target = null //A secondary target only used when we are not aggroing for a target we attack
+	var/mood = HOG_SKITTISH
+	var/panic = 0 //Ticks down, when positive cause aggro
+
+/mob/living/simple_animal/hostile/spacehog/death(var/gibbed = FALSE)
+	..(gibbed)
+	playsound(src, 'sound/effects/box_scream.ogg', 100, 1)
+
+/mob/living/simple_animal/hostile/spacehog/Process_Spacemove(var/check_drift = 0)
+	return 1 //All spacehogs are proficient in space navigation
+
+/mob/living/simple_animal/hostile/spacehog/before_retreat() //Before we take a step away, if the most direct route is blocked and the enemy is near, panic
+	var/target_distance = get_dist(src,target)
+	if(target_distance <= 2)
+		var/turf/T = get_step(src, get_dir(target,src)) //calculate the next space
+		if(T.density)
+			panic += 2
+			return
+		for(var/atom/movable/AM in T)
+			if(AM.density)
+				panic += 2
+				return
+
+/mob/living/simple_animal/hostile/spacehog/proc/set_mood(var/mood)
+	switch(mood)
+		if(HOG_CURIOUS)
+			idle_vision_range = 4
+			retreat_distance = 4
+			minimum_distance = 4
+			turns_per_move = 1
+		if(HOG_SKITTISH)
+			idle_vision_range = 8
+			retreat_distance = 8
+			minimum_distance = 8
+			turns_per_move = 5
+		if(HOG_HIT_AND_RUN)
+			idle_vision_range = 8
+			retreat_distance = 2
+			minimum_distance = 1
+			turns_per_move = 5
+		if(HOG_ASSAULT)
+			idle_vision_range = 8
+			retreat_distance = null
+			minimum_distance = 1
+			turns_per_move = 5
+
+/mob/living/simple_animal/hostile/spacehog/Life()
+	..()
+	nutrition--
+
+	if(stance == HOSTILE_STANCE_IDLE && (!food_target || get_dist(food_target, src) > 7) && !ckey)
+		if(mood == HOG_CURIOUS)
+			var/odds_of_new_food_search = max(300 - nutrition, 0)
+			//every nutrition under HOG_FED increases odds of a new search by 1%
+			if(prob(odds_of_new_food_search))
+				mood = HOG_SKITTISH
+			return
+		food_target = idle_search()
+		if(!food_target)
+			set_mood(HOG_CURIOUS)
+
+/* Priorities:
+planted seeds with ligneous = FALSE
+fresh meat on ground or fish
+cabbage on ground
+other loose veggies on ground or eggs, fish eggs, borer eggs
+other snacks
+passive animals, preferring smaller ones
+carrion
+if ungreased adult: l containers
+*/
+//share info across all hogs?
+
+/mob/living/simple_animal/hostile/spacehog/proc/idle_search()
+	var/tree_tray = null
+	for(var/obj/machinery/portable_atmospherics/hydroponics/H in view(food_search_radius,src))
+		if(H.seed)
+			if(H.seed.ligneous)
+				tree_tray = H
+			else
+				return H
+	for(var/obj/item/weapon/reagent_containers/food/snacks/meat/M in view(food_search_radius,src))
```

Comment:
Bit weird to call view() so many times but it's been optimised quite a lot so probably ok